pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -583,6 +583,7 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo, provisionC
 			"",
 		))
 
+	envVars = append(envVars, controller.ApplyNetworkEnv(&c.spec)...)
```

Comment:
please explain in the comment that it will be used to re-start the daemons

Reply:
Adding a comment here can be a little miss guidance as it is not just the env settings change that result in a restart but any change in spec of ceph pods will result in restart. So, I'll suggest let's skip adding comments.